internal/services/synapse/synapse_workspace_resource.go

```diff
@@ -68,9 +68,10 @@ func resourceSynapseWorkspace() *pluginsdk.Resource {
 			"location": commonschema.Location(),
 
 			"storage_data_lake_gen2_filesystem_id": {
-				Type:     pluginsdk.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IsURLWithPath,
```

Comment:
this is a resource id? should we not be validating it as an id not a uri?

Reply:
@katbyte, the ID here is in the form of a URI (e.g., `https://storageaccountname.dfs.core.windows.net/filesystemname`)